sdcclient/_secure.py

```diff
@@ -675,7 +768,152 @@ def delete_policy_id(self, id):
+        return self._request_result(res)
+
+    def delete_compliance_task(self, id):
+        '''**Description**
+            Delete the compliance task with the given id
+
+        **Arguments**
+            - id: the id of the compliance task to delete
+        '''
+        res = requests.delete(self.url + '/api/complianceTasks/{}'.format(id), headers=self.hdrs, verify=self.ssl_verify)
         if not self._checkResponse(res):
-            return [False, self.lasterr]
+            return False, self.lasterr
 
-        return [True, res.json()]
+        return True, None
+
+    def list_compliance_results(self, limit=50, direction=None, cursor=None, filter=""):
+        '''**Description**
+            Get the list of all compliance tasks runs.
+
+        **Arguments**
+            - limit: Maximum number of alerts in the response.
+            - direction: the direction (PREV or NEXT) that determines which results to return in relation to cursor.
+            - cursor: An opaque string representing the current position in the list of alerts. It's provided in the 'responseMetadata' of the list_alerts response.
+            - filter: an optional case insensitive filter used to match against the completed task name and return matching results.
+
+        **Success Return Value**
+            A JSON list with the representation of each compliance task run.
+        '''
+        url = "{url}/api/complianceResults?cursor{cursor}&filter={filter}&limit={limit}{direction}".format(
+            url=self.url,
+            limit=limit,
+            direction="&direction=%s" % direction if direction else "",
+            cursor="=%d" % cursor if cursor is not None else "",
+            filter=filter)
+        res = requests.get(url, headers=self.hdrs, verify=self.ssl_verify)
+        return self._request_result(res)
+
+    def get_compliance_results(self, id):
+        '''**Description**
+            Retrieve the details for a specific compliance task run result.
+
+        **Arguments**
+            - id: the id of the compliance task run to get.
+
+        **Success Return Value**
+            A JSON representation of the compliance task run result.
+        '''
+        res = requests.get(self.url + '/api/complianceResults/{}'.format(id), headers=self.hdrs, verify=self.ssl_verify)
+        return self._request_result(res)
+
+    def get_compliance_results_csv(self, id):
+        '''**Description**
+            Retrieve the details for a specific compliance task run result in csv.
+
+        **Arguments**
+            - id: the id of the compliance task run to get.
+
+        **Success Return Value**
+            A CSV representation of the compliance task run result.
+        '''
+        res = requests.get(self.url + '/api/complianceResults/{}/csv'.format(id), headers=self.hdrs, verify=self.ssl_verify)
+        if not self._checkResponse(res):
+            return False, self.lasterr
+
+        return True, res.text
```

Comment:
May we use here `self._request_result(res)` ?

Reply:
`self._request_result` in case of success returns `res.json`, in this case the return value is `res.text`.  I could make `self._request_result` more generic and allow returning other formats besides json, but for the few cases that differ I thought was not so worth it.

Reply:
Oh, did not check that.